app/aws/metrics.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations  # PEP 563 -- Postponed Evaluation of Annotations
+
+from typing import TYPE_CHECKING
+
+from aws_embedded_metrics import metric_scope  # type: ignore
+from aws_embedded_metrics.config import get_config  # type: ignore
+from botocore.exceptions import ClientError
+from flask import current_app
+
+from app.config import Config
+
+if TYPE_CHECKING:  # A special Python 3 constant that is assumed to be True by 3rd party static type checkers
+    from aws_embedded_metrics.logger.metrics_logger import MetricsLogger  # type: ignore
+
+    from app.queue import RedisQueue
+
+Metrics_Config = get_config()
+Metrics_Config.agent_endpoint = Config.CLOUDWATCH_AGENT_ENDPOINT
+Metrics_Config.service_name = "BatchSaving"
+Metrics_Config.service_type = "Redis"
+Metrics_Config.log_group_name = "BatchSaving"
```

Comment:
Would that run fine if ran in an environment without a cloudwatch agent endpoint? Modules/objects in Notify follow a pattern of bootstrapping configuration via an `init_app` method from the [application startup over here](https://github.com/cds-snc/notification-api/blob/9461b26f9582bdea6fe7c68b3d360597e5eab9ba/app/__init__.py#L60) and an example [with Celery over there](https://github.com/cds-snc/notification-api/blob/ce27097cf39fb9615771e5d581e3abdcf0bc2668/app/celery/celery.py#L38).

Reply:
@jimleroyer We might be okay, but we are in the process of testing this to be sure. Do you want us to change the initialization regardless though?

Reply:
We tested this locally and there is not issue with invocating the cloud watch metric

Reply:
My concern was more if configuration is missing, what happens? Will devs setting it up locally will have to provide the endpoint to run Notify? If not, it's ok. Else I'd suggest we make it easy to disable it if the endpoint is missing.

Reply:
So looking at the local logs, regardless of the endpoint being configured or not we see the data is submitted but doesn't wait for an ack:
```
[2022-02-21 21:04:00,946: INFO/MainProcess] Received task: heartbeart-inbox-sms[4c2178c2-3b8f-46ae-9cfa-22d5bb66e182]  
[2022-02-21 21:04:01,033: INFO/ForkPoolWorker-2] Parsed agent endpoint (udp) 127.0.0.1:8125
[2022-02-21 21:04:01,033: INFO/ForkPoolWorker-2] Submitted metrics to agent over UDP
```